test/e2e/main/scaling_test.go

```diff
@@ -9,79 +9,62 @@ import (
-func TestCheckDataSurviveToShutdown(t *testing.T) {
-	t.Parallel()
-	defer testKube.CleanNamespaceAndLogOnPanic(t, tutils.Namespace)
+	ispn := testKube.WaitForInfinispanCondition(spec.Name, spec.Namespace, ispnv1.ConditionWellFormed)
+	client_ := tutils.HTTPClientForCluster(ispn, testKube)
 
-	cacheName := "test"
-	template := `<infinispan><cache-container><distributed-cache name ="` + cacheName +
-		`"><persistence><file-store/></persistence></distributed-cache></cache-container></infinispan>`
-	key := "test"
-	value := "test-operator"
-
-	// Define function for the generic stop/start test procedure
-	var createCacheWithFileStore = func(ispn *ispnv1.Infinispan) {
-		client_ := tutils.HTTPClientForCluster(ispn, testKube)
-		cacheHelper := tutils.NewCacheHelper(cacheName, client_)
-		cacheHelper.Create(template, mime.ApplicationXml)
-		cacheHelper.Put(key, value, mime.TextPlain)
-	}
+	// Create non-persisted cache
+	defaultCacheHelper := tutils.NewCacheHelper(defaultCache, client_)
+	defaultCacheHelper.Create(defaultCacheConfig, mime.ApplicationXml)
 
-	var useCacheWithFileStore = func(ispn *ispnv1.Infinispan) {
-		client_ := tutils.HTTPClientForCluster(ispn, testKube)
-		cacheHelper := tutils.NewCacheHelper(cacheName, client_)
-		actual, _ := cacheHelper.Get(key)
-		if actual != value {
-			panic(fmt.Errorf("unexpected actual returned: %v (value %v)", actual, value))
-		}
-		cacheHelper.Delete()
-	}
+	// Create persisted cache with an entry
+	filestoreKey := "testFilestoreKey"
+	filestoreValue := "testFilestoreValue"
 
-	genericTestForGracefulShutdown(t, createCacheWithFileStore, useCacheWithFileStore)
-}
+	filestoreCacheHelper := tutils.NewCacheHelper(filestoreCache, client_)
+	filestoreCacheHelper.Create(filestoreCacheConfig, mime.ApplicationXml)
+	filestoreCacheHelper.Put(filestoreKey, filestoreValue, mime.TextPlain)
 
-func genericTestForGracefulShutdown(t *testing.T, modifier func(*ispnv1.Infinispan), verifier func(*ispnv1.Infinispan)) {
-	// Create a resource without passing any config
-	// Register it
-	spec := tutils.DefaultSpec(t, testKube)
-	testKube.CreateInfinispan(spec, tutils.Namespace)
-	testKube.WaitForInfinispanPods(1, tutils.SinglePodTimeout, spec.Name, tutils.Namespace)
-	ispn := testKube.WaitForInfinispanCondition(spec.Name, spec.Namespace, ispnv1.ConditionWellFormed)
+	// Shutdown/bring back the cluster
+	testKube.GracefulShutdownInfinispan(spec)
+	testKube.GracefulRestartInfinispan(spec, int32(replicas), tutils.SinglePodTimeout)
 
-	// Do something that needs to be permanent
-	modifier(ispn)
+	// Verify non-persisted cache usability
+	defaultKey := "defaultKey"
+	defaultValue := "defaultValue"
 
-	// Delete the cluster
-	testKube.GracefulShutdownInfinispan(spec)
-	testKube.GracefulRestartInfinispan(spec, 1, tutils.SinglePodTimeout)
+	defaultCacheHelper.TestBasicUsage(defaultKey, defaultValue)
+
+	// Server may return 500 upon deleting cache after graceful shutdown of a cluster
+	// defaultCacheHelper.Delete()
```

Comment:
This doesn't sound right, any idea why a 500 response is being returned?

Reply:
No idea, it doesn't happen always but often enough. There is an issue reported already.

Between keeping `deleteCache` and increasing number of cluster nodes to two I came to a conclusion that the later is more important to be covered by tests.

Reply:
> There is an issue reported already.

Do you have a link? I can't find it

Reply:
[ISPN-13740](https://issues.redhat.com/browse/ISPN-13740)

Reply:
Added Jira ID to the comment